### PR TITLE
Enhance LLDP network documentation

### PR DESCRIPTION
**Pull request by jukkar**
This requires still some updates after #9691 has been merged. So the current plan is to merge this one after #9691 is merged.

Edit: Change of plans. I will send any changes needed to LLDP documentation If the #9691 does make it, so we do not need to wait for it.

### CODE REVIEW

File: doc/reference/networking/lldp.rst [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-####
+Link Layer Discovery Protocol
+#############################
 
 Overview
 ********
 
+The Link Layer Discovery Protocol (LLDP) is a vendor-neutral link layer
+protocol used by network devices for advertising their identity, capabilities,
+and neighbors on an wired Ethernet.
```
Comment (dbkinder) [inline doc/reference/networking/lldp.rst]:
```suggestion
and neighbors on a wired Ethernet network.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
small tweak...

**Review by pfalcon — APPROVED** [review]
+1 once typos/build is fixed.

**Review by dbkinder — APPROVED** [review]

**Comment by jukkar** [comment]
Fixed sanity issues.

**Pull request merged**